### ISSUE TITLE
Exception thrown when using param, but not when using literal [Neo.DatabaseError.Statement.ExecutionFailed: next on empty iterator]

### ISSUE THREAD

**Issue by VinzentJenner** [neo4j/neo4j#13035]
An exception is thrown when using a param, but not explicitly stating literal. In the below query, where $age is set to a value, throws an exception.
```
MATCH (p:Person) -[var0]-> (var1:Food)
WHERE 
(type(var0) IN [] OR ((var1)-[:CONTAINS]->(:Ingredient) AND type(var0) IN ["EATS_FOOD"])) AND
p.age = $age
RETURN DISTINCT p
```
`Neo.DatabaseError.Statement.ExecutionFailed: next on empty iterator`

However, when avoiding using params, the query does not fail and returns the expected results.

```
MATCH (p:Person) -[var0]-> (var1:Food)
WHERE 
(type(var0) IN [] OR ((var1)-[:CONTAINS]->(:Ingredient) AND type(var0) IN ["EATS_FOOD"])) AND
p.age = 10
RETURN DISTINCT p
```

By removing the `OR` statement the issue is resolved. It seems the issue is related to the combination of the added `(var1)-[:CONTAINS]->(:Ingredient)` and the use of a parameter. 

--- 

Version: 4.4.11
Edition: Enterprise
Tested: Both using C# and the Browser GUI
Run on local kubernetes cluster

**Comment by HannesSandberg** [comment]
Thank you for your report! We will investigate and report back to you.

**Comment by hedengran** [comment]
Hi @VinzentJenner, thank you again for reporting this! I've investigated and this was indeed a bug. It turns out that for this oddly specific combination of predicates (pattern predicate, `IN`-predicate with empty list, and some predicate with a parameter), the planner could end up in a faulty state, which would be the source of the exception. I've fixed this in 4.4 and you should see the correct behavior with the next patch. Later version (5.X) was not affected by this bug.

**Comment by VinzentJenner** [comment]
Great, thanks for the quick response and fix! @HannesSandberg @hedengran
